@brb Travis CI failed:
```
-- PASS: TestBPFMapSizeCalculation (0.00s)
    --- PASS: TestBPFMapSizeCalculation/static_default_sizes (0.00s)
    --- PASS: TestBPFMapSizeCalculation/static,_non-default_sizes_inside_range (0.00s)
    --- PASS: TestBPFMapSizeCalculation/dynamic_size_without_any_static_sizes_(512MB,_0.25%) (0.00s)
    --- PASS: TestBPFMapSizeCalculation/dynamic_size_without_any_static_sizes_(1GiB,_0.25%) (0.00s)
    --- PASS: TestBPFMapSizeCalculation/dynamic_size_without_any_static_sizes_(2GiB,_0.25%) (0.00s)
    --- PASS: TestBPFMapSizeCalculation/dynamic_size_without_any_static_sizes_(7.5GiB,_0.25%) (0.00s)
    --- PASS: TestBPFMapSizeCalculation/dynamic_size_without_any_static_sizes_(16GiB,_0.25%) (0.00s)
    --- PASS: TestBPFMapSizeCalculation/dynamic_size_without_any_static_sizes_(120GiB,_0.25%) (0.00s)
    --- PASS: TestBPFMapSizeCalculation/dynamic_size_without_any_static_sizes_(240GiB,_0.25%) (0.00s)
    --- PASS: TestBPFMapSizeCalculation/dynamic_size_without_any_static_sizes_(360GiB,_0.25%) (0.00s)
    --- PASS: TestBPFMapSizeCalculation/dynamic_size_with_static_CT_TCP_size_(4GiB,_0.25%) (0.00s)
    --- PASS: TestBPFMapSizeCalculation/huge_dynamic_size_ratio_gets_clamped_(8GiB,_98%) (0.00s)
=== RUN   Test
/tmp/check-5577006791947779410/1/test
OK: 26 passed
--- PASS: Test (0.02s)
FAIL
coverage: 5.6% of statements in ./...
FAIL	github.com/cilium/cilium/pkg/option	0.100s
FAIL
Makefile:217: recipe for target 'unit-tests' failed
make: *** [unit-tests] Error 1
The command "./.travis/build.sh" exited with 2.

```

--- a/pkg/option/config.go
+++ b/pkg/option/config.go
@@ -2678,9 +2678,13 @@ func (c *DaemonConfig) checkMapSizeLimits() error {
 }
 
 func (c *DaemonConfig) checkIPv4NativeRoutingCIDR() error {
-	if c.IPv4NativeRoutingCIDR() == nil && c.Masquerade && c.Tunnel == TunnelDisabled && c.IPAMMode() != ipamOption.IPAMENI {
-		return fmt.Errorf("native routing cidr must be configured with option --%s in combination with --%s --%s=%s --%s=%s",
-			IPv4NativeRoutingCIDR, Masquerade, TunnelName, c.Tunnel, IPAM, c.IPAMMode())
+	if c.IPv4NativeRoutingCIDR() == nil && c.Masquerade && c.Tunnel == TunnelDisabled &&
+		c.IPAMMode() != ipamOption.IPAMENI && c.EnableIPv4 {
+		return fmt.Errorf(
+			"native routing cidr must be configured with option --%s "+
+				"in combination with --%s --%s=%s --%s=%s --%s=true",
+			IPv4NativeRoutingCIDR, Masquerade, TunnelName, c.Tunnel,
+			IPAM, c.IPAMMode(), EnableIPv4Name)
 	}
 
 	return nil

--- a/pkg/option/config_test.go
+++ b/pkg/option/config_test.go
@@ -494,6 +494,7 @@ func TestCheckIPv4NativeRoutingCIDR(t *testing.T) {
 				Tunnel:                TunnelDisabled,
 				IPAM:                  ipamOption.IPAMAzure,
 				ipv4NativeRoutingCIDR: cidr.MustParseCIDR("10.127.64.0/18"),
+				EnableIPv4:            true,
 			},
 			wantErr: false,
 		},
@@ -503,6 +504,7 @@ func TestCheckIPv4NativeRoutingCIDR(t *testing.T) {
 				Masquerade: false,
 				Tunnel:     TunnelDisabled,
 				IPAM:       ipamOption.IPAMAzure,
+				EnableIPv4: true,
 			},
 			wantErr: false,
 		},
@@ -512,6 +514,7 @@ func TestCheckIPv4NativeRoutingCIDR(t *testing.T) {
 				Masquerade: true,
 				Tunnel:     TunnelVXLAN,
 				IPAM:       ipamOption.IPAMAzure,
+				EnableIPv4: true,
 			},
 			wantErr: false,
 		},
@@ -521,6 +524,7 @@ func TestCheckIPv4NativeRoutingCIDR(t *testing.T) {
 				Masquerade: true,
 				Tunnel:     TunnelDisabled,
 				IPAM:       ipamOption.IPAMENI,
+				EnableIPv4: true,
 			},
 			wantErr: false,
 		},
@@ -530,6 +534,7 @@ func TestCheckIPv4NativeRoutingCIDR(t *testing.T) {
 				Masquerade: true,
 				Tunnel:     TunnelDisabled,
 				IPAM:       ipamOption.IPAMAzure,
+				EnableIPv4: true,
 			},
 			wantErr: true,
 		},
@@ -543,6 +548,7 @@ func TestCheckIPv4NativeRoutingCIDR(t *testing.T) {
 			}
 		})
 	}
+
 }
 
 func Test_populateNodePortRange(t *testing.T) {

